Mathlib/Algebra/Parity.lean

```diff
@@ -63,8 +63,40 @@ theorem isSquare_op_iff (a : α) : IsSquare (op a) ↔ IsSquare a :=
 #align is_square_op_iff isSquare_op_iff
 #align even_op_iff even_op_iff
 
+@[to_additive]
+theorem isSquare_unop_iff {a : αᵐᵒᵖ} : IsSquare (unop a) ↔ IsSquare a := by
+  rw [←isSquare_op_iff, op_unop]
+
+@[to_additive]
+instance [DecidablePred (@IsSquare α _)] : DecidablePred (@IsSquare (αᵐᵒᵖ) _) :=
```

Comment:
```suggestion
instance [@DecidablePred α IsSquare] : @DecidablePred αᵐᵒᵖ IsSquare :=
```

Reply:
What is the reason to prefer one over the other?

IMO we should be using the fancy new Lean 4 syntax:
```suggestion
instance [DecidablePred (α := α) IsSquare] : DecidablePred (α := αᵐᵒᵖ) IsSquare :=
```

Reply:
I went for a third option without `@` or specifying an argument name.